UiTests/Common/UiTestHelpers.cs

```diff
@@ -0,0 +1,572 @@
+                Thread.Sleep(2000);
+            }
+
+            //Verify that processes are running
+            for (int i = 0; i < numRetries; i++)
+            {
+                if (!UiTestHelpers.ProcessesAreAlive(processes.Values.ToList())) { RestartProcesses(processes, processDataEntries); }
+                else { break; }
+            }
+
+            if (!UiTestHelpers.ProcessesAreAlive(processes.Values.ToList()))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private static void RestartProcesses(Dictionary<string, Process> processes, List<ProcessStartOptions> processDataEntries)
+        {
+            //attempt to restart failed processes
+            foreach (KeyValuePair<string, Process> processEntry in processes)
+            {
+                if (!ProcessIsAlive(processEntry.Value))
+                {
+                    var processDataEntry = processDataEntries.Where(x => x.ExecutableName == processEntry.Key).Single();
+                    var process = StartProcessLocally(
+                                                    processDataEntry.TestAssemblyLocation,
+                                                    processDataEntry.AppLocation,
+                                                    processDataEntry.ExecutableName,
+                                                    processDataEntry.EnvironmentVariables);
+                    Thread.Sleep(5000);
+
+                    //Update process in collection
+                    processes[processEntry.Key] = process;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns a string representation of the running processes
+        /// </summary>
+        /// <param name="processes">Dict of running processes</param>
+        /// <returns>A string of all processes from the give dict</returns>
+        public static string GetRunningProcessAsString(Dictionary<string, Process>? processes)
+        {
+            StringBuilder runningProcesses = new StringBuilder();
+            if (processes != null)
+            {
+                foreach (var process in processes)
+                {
+#pragma warning disable CA1305 // Specify IFormatProvider
+                    runningProcesses.AppendLine($"Is {process.Key} running: {UiTestHelpers.ProcessIsAlive(process.Value)}");
+#pragma warning restore CA1305 // Specify IFormatProvider
+                }
+            }
+            return runningProcesses.ToString();
+        }
+
+        /// <summary>
+        /// Takes two paths to existing files and swaps their names and locations effectively swapping the contents of the files.
+        /// </summary>
+        /// <param name="path1">The path of the first file to swap</param>
+        /// <param name="path2">The path of the file to swap it with</param>
+        private static void SwapFiles(string path1, string path2)
```

Comment:
Should there be any error handling for file IO exceptions?

Reply:
Added some.